gemviz23/demo/resultwindow.py

```diff
@@ -33,24 +48,17 @@ def columnCount(self, parent=None):
     def data(self, index, role=None):
         # display data
         if role == QtCore.Qt.DisplayRole:
-            #print("Display role:", index.row(), index.column())
-            uid=self.uidList()[index.row()]
-            run=self.catalog()[uid]
-            column=index.column()
-            if column==0:
-                return self._getKey(run, "start", "scan_id")
-            elif column==1:
-                return self._getKey(run, "start", "plan_name")
-            elif column==2:
-                return ", ".join(self._getKey(run,"start", "motors", []))
-            elif column==3:
-                return ", ".join(self._getKey(run,"start", "detectors", []))
-            elif column==4:
-                ts = self._getKey(run, "start", "time")
-                dt = datetime.datetime.fromtimestamp(round(ts))
-                return dt.isoformat(sep=" ")
-            elif column==5:
-                return self._getKey(run, "stop", "exit_status")
+            # print("Display role:", index.row(), index.column())
+            uid = self.uidList()[index.row()]
+            run = self.catalog()[uid]
+
+            label = self.columnLabels[index.column()]
+            action = self.actions_library[label]
+
+            if isinstance(action, list):
+                return self.get_md(run, *action)
+            else:
+                return action(run)
             
     def _getKey(self, run, document_name, key, default=""):
```

Comment:
The _getKey method is no longer used in this case, is it?

Reply:
That's right.  It was removed in a later commit.